### DISCUSSION TITLE
Verify without LoginForm

### DISCUSSION THREAD

**Discussion by meteoDaniel** [fief-dev/fief#374] [Ideas]
Dear fief-Team, 

I really like your porject and I started to dig deeper into it because I really need a User Management tool for our applications based upon FastAPI. 

As far as I understand, Login is only possible via the LoginForm in the Browser. 

https://github.com/fief-dev/fief/blob/9c7e72e4d37c078c722a5445bc12a6c1e159b274/fief/apps/auth/routers/auth.py#L154

So for me it would be very helpful to have one of the two options: 
1. Login made available via query/header/cookies by passing E-Mail and Password to retrieve a Token
2. User can generate an Access Token in its User Profil similar to the way it works in the admin dashboard for the admin. 

A possible solution can look like this: 
https://fastapi.tiangolo.com/tutorial/security/simple-oauth2/

If you can tell me where to add a first guess, I can open-up a PR to support the dev of the feature.

**Comment by meteoDaniel** [comment]
PS: I have found a work around that will work for me so far. But really looking forward to have such feature in fief :)

**Comment by frankie567** [comment]
Yes, for now login is only handled through our own UI. There is no API allowing you to "integrate" this process in your own app.

This is something we'll probably add in the future, but it's a quite complex one (we have to handle lot different use cases, like social login, email verification or even 2FA).
